plugins/inputs/prometheus/consul.go

```diff
@@ -0,0 +1,174 @@
+package prometheus
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"net/url"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+)
+
+type ConsulServiceInfo struct {
+	ID             string
+	Node           string
+	Address        string
+	Datacenter     string
+	NodeMeta       map[string]string
+	ServiceID      string
+	ServiceName    string
+	ServiceAddress string
+	ServiceTags    []string
+	ServiceMeta    map[string]string
+	ServicePort    int
+}
+
+func convertConsulService(s *api.CatalogService) *ConsulServiceInfo {
+	return &ConsulServiceInfo{
+		ID:             s.ID,
+		Node:           s.Node,
+		Address:        s.Address,
+		Datacenter:     s.Datacenter,
+		NodeMeta:       s.NodeMeta,
+		ServiceID:      s.ServiceID,
+		ServiceAddress: s.ServiceAddress,
+		ServiceTags:    s.ServiceTags,
+		ServiceMeta:    s.ServiceMeta,
+		ServicePort:    s.ServicePort,
+	}
+}
+
+func (p *Prometheus) startConsul(ctx context.Context) error {
+	config := api.DefaultConfig()
+	if p.ConsulConfig.Agent != "" {
+		config.Address = p.ConsulConfig.Agent
+	}
+
+	consul, err := api.NewClient(config)
+	if err != nil {
+		log.Printf("E! [inputs.prometheus] Cannot connect to the Consul agent. Error: %s", err)
+		return nil
+	}
+
+	// Parse the template for metrics URL
+	for i, q := range p.ConsulConfig.Queries {
+		serviceURLTemplate, err := template.New("URL").Parse(q.ServiceURL)
+		if err != nil {
+			log.Printf("E! [inputs.prometheus] Could not parse the Consul query URL template (%s), skipping it. Error: %s", q.ServiceURL, err)
```

Comment:
Please use
```suggestion
			p.Log.Errorf("Could not parse the Consul query URL template (%s), skipping it. Error: %s", q.ServiceURL, err)
```
instead!